pkg/maps/policymap/policymap.go

```diff
@@ -441,5 +446,21 @@ func InitCallMap() error {
 		bpf.ConvertKeyValue,
 	)
 	_, err := policyCallMap.Create()
+
+	if err == nil && option.Config.EnableEnvoyConfig {
+		policyEgressCallMap := bpf.NewMap(PolicyEgressCallMapName,
+			bpf.MapTypeProgArray,
+			&CallKey{},
+			int(unsafe.Sizeof(CallKey{})),
+			&CallValue{},
+			int(unsafe.Sizeof(CallValue{})),
+			int(PolicyCallMaxEntries),
+			0,
+			0,
+			bpf.ConvertKeyValue,
+		)
+
+		_, err = policyEgressCallMap.Create()
+	}
 	return err
```

Comment:
Same here, let's try and move this to the caller instead.

Reply:
You mean passing in a bool for `option.Config.EnableEnvoyConfig`??

Reply:
Either that or pass an interface something like it's being done [here](https://github.com/cilium/cilium/blob/f0465215aec02e7c21887d672840f6ff03780374/pkg/service/store/store.go#L162-L188)

Reply:
Done.